hash-tables/overview-of-hash-tables.md

```diff
@@ -0,0 +1,153 @@
+# Overview of Hash Tables
+
+## Learning Goals
+
+- Define hash table
+- Define hash function
+- Describe how hash tables resolve collisions
+
+## Introduction
+
+Python dictionaries are fascinating! They store key-value pairs, and, as long as we have the key, getting the value has a time complexity of O(1).
+
+This is in direct comparison to arrays/lists. We could store elements in an array, but finding specific entries in an array requires looping through potentially the entire list an O(n) operation.
+
+How do Python dictionaries achieve this speed? How do keys and values get allocated memory? Under the hood, Python dictionaries are _hash tables_.
```

Comment:
some minor edits, and split into two sections to separate the question and answer for a little drama.
```suggestion
How do Python dictionaries achieve this performance? How are keys and values allocated in memory?

Under the hood, Python dictionaries are _hash tables_.
```